community/modules/scheduler/cloud-batch-login-node/main.tf

```diff
@@ -32,19 +46,20 @@ locals {
 
   login_metadata = merge(local.instance_template_metadata, local.startup_metadata, local.oslogin_metadata)
 
-  batch_command_instructions = <<-EOT
+  batch_command_instructions = join("\n", [for job in local.job_data : <<-EOT
+  ## For job: ${job.id} ##
+
   Submit your job from login node:
-    gcloud ${var.gcloud_version} batch jobs submit ${var.job_id} --config=${local.job_template_destination} --location=${var.region} --project=${var.project_id}
+    gcloud ${var.gcloud_version} batch jobs submit ${job.id} --config=${var.batch_job_directory}/${job.filename} --location=${var.region} --project=${var.project_id}
   
   Check status:
-    gcloud ${var.gcloud_version} batch jobs describe ${var.job_id} --location=${var.region} --project=${var.project_id} | grep state:
+    gcloud ${var.gcloud_version} batch jobs describe ${job.id} --location=${var.region} --project=${var.project_id} | grep state:
   
   Delete job:
-    gcloud ${var.gcloud_version} batch jobs delete ${var.job_id} --location=${var.region} --project=${var.project_id}
+    gcloud ${var.gcloud_version} batch jobs delete ${job.id} --location=${var.region} --project=${var.project_id}
 
-  List all jobs:
-    gcloud ${var.gcloud_version} batch jobs list --project=${var.project_id}
   EOT
+  ])
 
   readme_contents = <<-EOT
```

Comment:
nit: now readme contents will not contain the List all jobs command. Probably worth including but not critical.

Reply:
It still does include the list all jobs command - it just moved to the top. This is the output from the test_config:

```
Batch job template files will be placed on the Batch login node in the following directory:
          /home/batch-jobs
        
        Use the following commands to:
        SSH into the login node:
          gcloud compute ssh --zone us-central1-c hello-workload-batch-login  --project gcp-hpc-demo-0
        
        List all jobs:
          gcloud alpha batch jobs list --project=gcp-hpc-demo-0
        
        ## For job: hello-workload ##
        
        Submit your job from login node:
          gcloud alpha batch jobs submit hello-workload --config=/home/batch-jobs/cloud-batch-hello-workload.json --location=us-central1 --project=gcp-hpc-demo-0
          
        Check status:
          gcloud alpha batch jobs describe hello-workload --location=us-central1 --project=gcp-hpc-demo-0 | grep state:
          
        Delete job:
          gcloud alpha batch jobs delete hello-workload --location=us-central1 --project=gcp-hpc-demo-0
        
        
        ## For job: job-2 ##
        
        Submit your job from login node:
          gcloud alpha batch jobs submit job-2 --config=/home/batch-jobs/batch-job-2.json --location=us-central1 --project=gcp-hpc-demo-0
          
        Check status:
          gcloud alpha batch jobs describe job-2 --location=us-central1 --project=gcp-hpc-demo-0 | grep state:
          
        Delete job:
          gcloud alpha batch jobs delete job-2 --location=us-central1 --project=gcp-hpc-demo-0
```

Reply:
I meant the readme that is placed on the login node by the data runner ([ref](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/372fe04b7d7a37dba5d5027d11b4d5c6680f1bc4/community/modules/scheduler/cloud-batch-login-node/main.tf#L107)). I believe the text you posted above is the instructions output.

Reply:
Understood, I got it confused with the instructions. I converted the line to a local var and used in both.